### PR TITLE
(feat): page-preview

### PR DESCRIPTION
**Pull request by Palmaswell**
Implementation of the page preview

### CODE REVIEW

File: src/component/page-list/page-tile-container.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		}, 300);
+	}
+
+	protected handleDoubleClick(e: React.MouseEvent<HTMLElement>): void {
+		Store.getInstance().openPage(this.props.page.getId());
+	}
+
+	@MobX.action
+	protected handleKeyDown(e: React.KeyboardEvent<HTMLInputElement>): void {
+		switch (e.key.toString()) {
```
Comment (LKuechler) [inline src/component/page-list/page-tile-container.tsx]: `e.key` should already be a string or am i missing something?

File: src/component/page-list/page-tile-composite.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+import { observer } from 'mobx-react';
+import { PreviewTile, PreviewTileProps } from '../../lsg/patterns/preview-tile';
+import Space, { Size } from '../../lsg/patterns/space';
+
+export const PageTileComposite: React.StatelessComponent<PreviewTileProps> = observer(
```
Comment (TheReincarnator) [inline src/component/page-list/page-tile-composite.tsx]: Why is this a React function and not a regular React class?

Comment (TheReincarnator) [inline src/component/page-list/page-tile-composite.tsx]: Does it make sense to have a composite component when its only purpose is a spacer? Much boiler-plate code here. Maybe put that composition code into the PreviewTile or container?

File: src/component/page-list/page-tile-container.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	@MobX.action
+	protected handleChange(e: React.ChangeEvent<HTMLInputElement>): void {
+		this.inputValue = e.target.value;
+	}
+
+	@MobX.action
+	protected handleClick(e: React.MouseEvent<HTMLElement>): void {
+		this.props.onClick(e);
+
+		setTimeout(() => {
```
Comment (TheReincarnator) [inline src/component/page-list/page-tile-container.tsx]: Timeout solutions are almost always dirty and incorrect ;-)

File: src/component/page-list/page-tile-container.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+@observer
+export class PageTileContainer extends React.Component<PageTileContainerProps> {
+	@MobX.observable public editable: boolean = false;
+	@MobX.observable public inputValue: string = '';
+
+	public constructor(props: PageTileContainerProps) {
+		super(props);
+		this.inputValue = this.inputValue ? this.inputValue : this.props.page.getName();
```
Comment (TheReincarnator) [inline src/component/page-list/page-tile-container.tsx]: What is the difference between this an
```
this.inputValue = this.props.page.getName()
```

File: src/component/page-list/page-tile-container.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+		this.handleChange = this.handleChange.bind(this);
+		this.handleClick = this.handleClick.bind(this);
+		this.handleDoubleClick = this.handleDoubleClick.bind(this);
+		this.handleKeyDown = this.handleKeyDown.bind(this);
+	}
+
+	@MobX.action
+	protected handleBlur(): void {
+		this.editable = false;
```
Comment (TheReincarnator) [inline src/component/page-list/page-tile-container.tsx]: Blur should commit the change, shouldn't it?

File: src/component/container/app.tsx [hunk 6]
```diff
@@ -65,6 +66,15 @@ export class App extends React.Component {
 		}
 	}
 
+	protected getLastChangedAuthor(): string {
+		return 'Max Mustermann';
```
[the remark this answers is not in the document]
Comment (LKuechler) [reply src/component/container/app.tsx]: The author should probably be removed completely. There is way to much missing for showing the author. This should just be done with a different story

File: src/component/page-list/page-tile-composite.tsx (hunk 2 above)
Comment (Palmaswell) [reply src/component/page-list/page-tile-composite.tsx]: You are right. The `page-tile-composite.tsx ` is gone :)

File: src/component/page-list/page-tile-container.tsx (hunk 4 above)
Comment (Palmaswell) [reply src/component/page-list/page-tile-container.tsx]: The `this.inputValue` is a dynamic value that changes with the entered value from the input element. But if the value has not been set, then it should use the current name of the page.

File: src/component/page-list/page-tile-container.tsx (hunk 5 above)
Comment (Palmaswell) [reply src/component/page-list/page-tile-container.tsx]: When the input element is no longer `:focus` it should just set the `this.editable` flag to false. Changes are not saved.

Is that what you mean @TheReincarnator?

File: src/component/page-list/page-list-preview.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import { Headline } from '../../lsg/patterns/headline';
+import Space, { Size } from '../../lsg/patterns/space';
+
+export interface PageListPreviewProps {
+	headline: string;
+	lastChangedDate: string;
+}
+
+export const PageListPreview: React.StatelessComponent<PageListPreviewProps> = props => (
+	<Space size={[Size.XXXL, Size.XL, Size.XS, Size.XL]}>
```
[the remark this answers is not in the document]
Comment (Palmaswell) [reply src/component/page-list/page-list-preview.tsx]: @LKuechler I actually do think it should be a LSG pattern. The look it's to specific to be a reusable and generic component. Until now only for the page preview.

File: src/component/page-list/page-tile-composite.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+import { observer } from 'mobx-react';
```
[the remark this answers is not in the document]
Comment (Palmaswell) [reply src/component/page-list/page-tile-composite.tsx]: The component has been removed. In this case, I guess it does not matter any longer 😃

File: src/component/page-list/page-tile-composite.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import { PreviewTile, PreviewTileProps } from '../../lsg/patterns/preview-tile';
+import Space, { Size } from '../../lsg/patterns/space';
+
+export const PageTileComposite: React.StatelessComponent<PreviewTileProps> = observer(
+	(props): JSX.Element => (
+		<Space size={Size.S}>
```
[the remark this answers is not in the document]
Comment (Palmaswell) [reply src/component/page-list/page-tile-composite.tsx]: Spacing is very much part of the composition. LSG patterns should remain as generic as possible. This means without any spaces too

File: src/component/page-list/page-tile-container.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Palmaswell) [reply src/component/page-list/page-tile-container.tsx]: Done 👍

File: src/component/page-list/page-tile-container.tsx (hunk 3 above)
Comment (Palmaswell) [reply src/component/page-list/page-tile-container.tsx]: @TheReincarnator you are right and it's gone

[the remark this answers is not in the document]
Comment (Palmaswell) [reply src/component/page-list/page-tile-container.tsx]: @LKuechler you are right and it's gone

File: src/component/page-list/page-tile-composite.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (LKuechler) [reply src/component/page-list/page-tile-composite.tsx]: Spacing is styling. If we go by this logic everything can be a component (eg. colors, fonts) and can be done in the composition step.

File: src/component/page-list/page-tile-container.tsx (hunk 5 above)
Comment (TheReincarnator) [reply src/component/page-list/page-tile-container.tsx]: Yes but at least on Windows, the common patterns is that blurring a rename field commits it, not aborts it. How does the Finder behave?

File: src/store/project.ts [hunk 10]
```diff
@@ -82,6 +82,16 @@ export class Project {
 		return this.id;
 	}
 
+	// TODO
+	public getLastChangedAuthor(): string {
```
Comment (Jumace) [inline src/store/project.ts]: We don't have any user management at the moment and afaik there is no plan to implement one while we're still on alpha. Based on that I would remove this function as it's now only dead code.

File: src/component/page-list/page-tile-container.tsx (hunk 5 above)
Comment (LKuechler) [reply src/component/page-list/page-tile-container.tsx]: In the finder the changes are saved on blur

File: src/store/project.ts [hunk 11]
```diff
@@ -82,6 +82,16 @@ export class Project {
...
+		return 'Max Mustermann';
+	}
+
+	// TODO
+	public getLastChangedDate(): Date {
```
Comment (Jumace) [inline src/store/project.ts]: I don't see that function will make sense at all, at least in the moment.
We will need to prepare more to make that work, which is also not planned at the moment. That's why I would delete that code too, it's dead and will not be used in the near future

File: src/store/project.ts [hunk 12]
```diff
@@ -51,6 +68,8 @@ export class Project {
 	public constructor(properties: ProjectProperties) {
 		this.id = properties.id ? properties.id : Uuid.v4();
 		this.name = properties.name;
+		this.lastChangedAuthor = properties.lastChangedAuthor || 'unknown';
```
Comment (Jumace) [inline src/store/project.ts]: in line 35 the 'lastChangedAuthor' is set to 'none' if it's not defined, here it's set to 'unknown', we should only do one of both

Comment (TheReincarnator) [reply src/store/project.ts]: Yes, you are right. I fixed this now.

Comment (TheReincarnator) [reply src/store/project.ts]: Done.

**Pull request merged**